docs/src/_tutorials/logging.md

```diff
@@ -0,0 +1,284 @@
+- `loggers` - This section allows you to explicitly control specific module/class/etc named loggers.
+
+A few key points to note:
+
+1. Different handlers can use different formats.
+2. Different loggers can use different handlers and log at different log levels.
+3. We support all the [standard python logging handlers](https://docs.python.org/3/library/logging.handlers.html#) (e.g. rotating files, syslog, etc).
+
+Here's an annotated example of a logging.yaml file:
+
+```yaml
+version: 1
+disable_existing_loggers: false
+
+formatters:
+  default: # use a format similar to default generic python logging format
+    format: "[%(asctime)s] [%(process)d|%(threadName)10s|%(name)s] [%(levelname)s] %(message)s"
+  structured_plain: # log format for structured plain text logs without colored output
+    (): meltano.core.logging.console_log_formatter
+    colors: False
+  structured_colored: # log format for structured plain text logs WITH colored output
+    (): meltano.core.logging.console_log_formatter
+    colors: True
+  key_value: # log format for traditional key=value style logs
+    (): meltano.core.logging.key_value_formatter
+    sort_keys: False
+  json: # log format for json formatted logs
+    (): meltano.core.logging.json_formatter
+
+handlers:
+  console: # log to the console (stderr) using structured_colored formatter, logging everything at DEBUG level and up
+    class: logging.StreamHandler
+    level: DEBUG
+    formatter: structured_colored
+    stream: "ext://sys.stderr"
+  meltano_log: # log everything INFO and above to a file in the project root called meltano.log in json format
+    class: logging.FileHandler
+    level: INFO
+    filename: meltano.log
+    formatter: json
+  my_warn_file_handler: # log everything WARNING and above to automatically rotating log file in key_value format
+    class: logging.handlers.RotatingFileHandler
+    level: WARN
+    formatter: key_value
+    filename: /tmp/meltano_warn.log
+    maxBytes: 10485760
+    backupCount: 20
+    encoding: utf8
+
+root:
+  level: DEBUG
+  propagate: yes # propagate to child loggers
+  handlers: [console, meltano_log, my_info_file_handler] # by default use these three handlers
+
+loggers:
+  somespecific.module.logger: # if you want debug logs for a specific named logger or module
+    level: DEBUG
+    handlers: [console]
+    propogate: no
+  urllib3: # for example hide all urllib3 debug logs
+    level: WARNING
+    handlers: [console, meltano_log]
+    propogate: no
+```
+
+## Local development config example
+
+While working with meltano locally it's sometimes nice to have more terse logging on the console, but still have DEBUG
+level info written to a log file behind scenes incase you need to debug something. To accomplish that, you can use a
+file like:
+
+```yaml
+version: 1
+disable_existing_loggers: false
+
+formatters:
+  structured_colored:
+    (): meltano.core.logging.console_log_formatter
```

Comment:
How does a user know the world of available options here?

Reply:
Doh good point, will update to document our current options.

Reply:
updated. pointed at the src file that holds our formatters and listed what they currently are.